exporter/containerimage/writer.go

```diff
@@ -783,19 +783,23 @@ func normalizeLayersAndHistory(ctx context.Context, remote *solver.Remote, histo
 	return remote, history
 }
 
-func RemoveInternalLayerAnnotations(desc *ocispecs.Descriptor, oci bool) {
-	if oci {
-		// oci supports annotations but don't export internal annotations
-		delete(desc.Annotations, "containerd.io/uncompressed")
-		delete(desc.Annotations, "buildkit/createdat")
-		for k := range desc.Annotations {
-			if strings.HasPrefix(k, "containerd.io/distribution.source.") {
-				delete(desc.Annotations, k)
+func RemoveInternalLayerAnnotations(desc ocispecs.Descriptor, oci bool) map[string]string {
```

Comment:
nit: signature can be `func (annotations map[string]string, oci bool) map[string]string`, instead of needing the full descriptor.